learning/01_Exercises/Lab 7 to 9/services/controller.py

```
@@ -0,0 +1,161 @@
+        self.validatorForClient = validatorForClient
+
+    # ADD SECTION
+    def addMovie(self, movie):
+        # Save the movie information in the list by validating the information given.
+        if self.validatorForMovie.validator(movie, self.moviesRepository.getNames()):
+            self.moviesRepository.save(movie)
+            print(f"\n{movie} has been added to the list.")
+        else:
+            print(f"The {movie} contains some wrong information")
+
+    def addClient(self, Client):
+        # Save the client's information in the list.
+        if self.validatorForClient.validator(Client, self.clientsRepository.getNames()):
+            self.clientsRepository.save(Client)
+            print(f"\n{Client} has been added to the list.")
+        else:
+            print(f"The {Client}'s details appears to be wrong, please try again.")
+
+    # REMOVE SECTION
+    def removeMovie(self, ID):
+        if ID in self.moviesRepository.getIdList():
+            newMovieList = self.moviesRepository.remove(ID)
+            self.moviesRepository.removeAll(), self.moviesRepository.saveAll(newMovieList)
+            print(f"\nThe movie with {ID} id has been removed.")
+        else:
+            print(f"\nThe ID: {ID} is invalid! Please try again.")
+
+    def removeClient(self, ID):
+        if ID in self.clientsRepository.getIdList():
+            newClientList = self.clientsRepository.remove(ID)
+            self.clientsRepository.removeAll(), self.clientsRepository.saveAll(newClientList)
+            print(f"\nThe client with {ID} id has been removed.")
+        else:
+            print(f"\nThe ID: {ID} is invalid! Please try again.")
+
+    # UPDATE SECTION
+    def updateMovie(self, ID, newMovie):
+        if self.validatorForMovie.checkID(ID, self.moviesRepository.getIdList()):
+            self.moviesRepository.update(ID, newMovie)
+        else:
+            print(f"\nThe ID: {ID} is invalid! Please try again.")
```

This should raise an error, Controller should not print to the console. The error should be handled in the ui

---

Rezolvat.